src/status_im/contexts/chat/messenger/placeholder/view.cljs

```diff
@@ -21,5 +22,8 @@
      (when-not chat-exist?
        [quo/page-nav
         {:icon-name :i/arrow-left
-         :on-press  #(rf/dispatch [:navigate-back])}])
+         :on-press  (fn []
+                      (rf/dispatch [:navigate-back])
+                      (when platform/ios?
+                        (rf/dispatch [:chat/close])))}])
```

Comment:
@Parveshdhull in the implementation of `:navigate-back` there's already code to handle Android:

https://github.com/status-im/status-mobile/blob/49231575d24f83597e5a91ed9989c68e831b072f/src/status_im/navigation/events.cljs#L54-L57

Shouldn't we handle iOS there in the event as well? Otherwise we risk a future developer likely forgetting to call `:chat/close`. From the snippet above, it seems that the Android check relies on the `:view-id`, whereas the placeholder detection uses the sub `:chats/current-chat-exist?`, which is a completely different implementation. Which approach is best? Seems odd we do two different types of detections, one for iOS and another for Android.

Reply:
> Shouldn't we handle iOS there in the event as well

hi @ilmotta, Thank you for reviewing the PR. Please check out PR description for more details related to this.

Reply:
> Seems odd we do two different types of detections, one for iOS and another for Android.

Yes, but both devices have different requirements. Only android supports hardware back button, and only ios have issue with swipe.

Reply:
I had read before, but I see now the connection/reason for the code to be like this. Still, why shouldn't we detect inside event `:navigate-back` that we should also dispatch `:chat/close` in iOS? The result of `(rf/sub [:chats/current-chat-exist?])` is a computation we could do inside event `:navigate-back` for iOS. Do you think that would be a bad approach, why?

Reply:
Makes complete sense. Thanks for the explanation